package.json

```diff
@@ -10,7 +10,7 @@
     "build": "webpack --config webpack.prod.js",
     "clean": "rimraf dist",
     "start:dev": "webpack serve --hot --color --progress --config webpack.dev.js",
-    "test": "jest",
+    "test": "jest --maxWorkers=1",
```

Comment:
I didn't really understand this - how/why does limiting the maxWorkers to 1 produce a performance speedup? That seems very counterintuitive

Reply:
[This article](https://dev.to/vantanev/make-your-jest-tests-up-to-20-faster-by-changing-a-single-setting-i36) does a comparison of various `--maxWorkers` settings while [this post](https://stackoverflow.com/questions/43864793/why-does-jest-runinband-speed-up-tests) attempts to answer your question. For the second link, note that `--runInBand` is equivalent to `--maxWorkers=1`. 

As the first article mentions, Jest defaults to using all available CPU threads but for some reason, this results in suboptimal performance. In my experience, it was taking a few minutes for just 2 tests. From the second link, it seems Jest creates a cache for each thread in the worker-pool which suggests that there's a performance tradeoff dependent on the thread:cache ratio. We also have the option of disabling the cache.

I went with `--maxWorkers=1` since others have found that to be the optimal setting for resource-constrained environments such as CI. However, it looks like `--maxWorkers=50%` could be a good setting for local tests so I could look into doing two separate settings for local and CI. We can tweak this setting as we add more tests and get a better picture of how long the test suite takes to run.

Reply:
> I went with --maxWorkers=1 since others have found that to be the optimal setting for resource-constrained environments such as CI. However, it looks like --maxWorkers=50% could be a good setting for local tests so I could look into doing two separate settings for local and CI. We can tweak this setting as we add more tests and get a better picture of how long the test suite takes to run.

I think local vs CI config makes a lot of sense too, I'd be curious to see how that works out.

Reply:
I tried out various `--maxWorkers` configurations and there wasn't any significant difference in speed during CI testing so I've settled on `--maxWorkers=50%` for both the local and CI configurations. Hopefully, as we add more tests this should provide a nice balance between the efficiency granted by threading and the overhead from using a separate cache for each thread. 

An interesting thing to note is that now when I run `jest` locally with the default maximum number of available threads, tests no longer takes minutes to complete. However, the `isolatedModules` setting in `jest.config.js` is still necessary to prevent the performance slowdown. I guess it's possible that fixing the dependencies resolved `--maxWorkers` involvement in the issue?

Reply:
If that does happen we do have the `--runInBand` option as a fallback so it shouldn't be a big issue.

Reply:
Anyways, I'll add the `TESTING.md` file as you requested then open the PR for the rest of the team to review as well.